### PR TITLE
Dont generate a popup notification for transactions from within metmask-extension 

### LINKED ISSUE #626: Do not open new popup when initiating tx from our UI
Blocked by #616

Once we can listen to popup close events properly, we'll able to monitor & respond appropriately here.

### PR DESCRIPTION
**Pull request by frankiebee**
fixes #626

### CODE REVIEW

**Comment by flyswatter** [comment]
Aw man, I should've waited 5 minutes before pushing that last version to the store!

**Comment by flyswatter** [comment]
I pulled the branch, and sending a tx from within the popup is still opening another popup.

**Comment by flyswatter** [comment]
I found that problem and fixed it @frankiebee

**Comment by flyswatter** [comment]
There's one other visual glitch, when first rejecting an internally generated tx, there's a flash of error text over the body before transitioning back. It's kindof ugly, would you like to try to fix that or should I?

File: app/scripts/background.js [hunk 1]
```diff
@@ -10,31 +10,22 @@ const MetamaskController = require('./metamask-controller')
...
+  showUnconfirmedMessage: triggerUi,
+  unlockAccountMessage: triggerUi,
+  showUnconfirmedTx: triggerUi,
   // Persistence Methods:
   setData,
   loadData,
 })
 const idStore = controller.idStore
 
-function unlockAccountMessage () {
```
Comment (flyswatter) [inline app/scripts/background.js]: I like all this cleanup and unification in `triggerUi`.

File: app/scripts/background.js [hunk 2]
```diff
@@ -53,7 +44,8 @@ function connectRemote (remotePort) {
   var portStream = new PortStream(remotePort)
   if (isMetaMaskInternalProcess) {
     // communication with popup
-    setupTrustedCommunication(portStream, 'MetaMask')
+    popupIsOpen = remotePort.name === 'popup'
```
Comment (flyswatter) [inline app/scripts/background.js]: We had this general assignment in two different methods which were redundant.

**Review by danfinlay — CHANGES_REQUESTED** [review]
Added a couple notes.  Using this new review feature now I have to summarize them?

Uh I already commented about a visual glitch I'd like us to address.

**Comment by flyswatter** [comment]
Whoah now marking 'changes requested' makes it not mergable!

I'm open to merging this anyways. I was just indicating that there's a visual glitch that I mentioned.

**Comment by frankiebee** [comment]
I can try and take care of the glitch

**Comment by flyswatter** [comment]
The glitch comes from how we show any errors we throw, and user rejection is an error of sorts, but I think the problem is that user rejection is a different kind of error that we shouldn't show ourselves, only pass back to the dapp.

File: app/scripts/background.js [hunk 3]
```diff
@@ -69,7 +61,7 @@ function setupUntrustedCommunication (connectionStream, originDomain) {
   controller.setupPublicConfig(mx.createStream('publicConfig'))
 }
 
-function setupTrustedCommunication (connectionStream, originDomain) {
+function setupTrustedCommunication (connectionStream, originDomain, metamaskContext) {
```
Comment (kumavis) [inline app/scripts/background.js]: @frankiebee remove this unused arg

**Review by kumavis — CHANGES_REQUESTED** [review]

**Review by danfinlay — APPROVED** [review]

File: app/scripts/background.js (hunk 3 above)
Comment (flyswatter) [inline app/scripts/background.js]: This change was made.

**Review by kumavis — APPROVED** [review]

**Pull request merged**